terraform/deployments/github/main.tf

```diff
@@ -164,6 +164,12 @@ resource "github_team_repository" "ithc_repos" {
   permission = try(each.value.teams["govuk_ithc"], "pull")
 }
 
+data "github_repository_pull_requests" "govuk_repos_prs" {
```

Comment:
You can probably for_each this based on only repos that need to be archived, that should massively reduce the queries made to github?